src/EnergyPlus/InputProcessing/InputProcessor.hh

```diff
@@ -335,7 +335,37 @@ private:
 
 struct DataInputProcessing : BaseGlobalStruct
 {
-    std::unique_ptr<InputProcessor> inputProcessor = InputProcessor::factory();
+    struct LazyConstructedInputProcessor
+    {
+        std::unique_ptr<InputProcessor> m_inputProcessor;
+        void reset()
+        {
+            m_inputProcessor.reset();
+        }
+
+        LazyConstructedInputProcessor &operator=(std::unique_ptr<InputProcessor> new_val)
+        {
+            m_inputProcessor = std::move(new_val);
+            return *this;
+        }
+
+        explicit operator bool() const noexcept
+        {
+            return static_cast<bool>(m_inputProcessor);
+        }
+
+        InputProcessor *operator->()
+        {
+            if (!m_inputProcessor) {
+                m_inputProcessor = EnergyPlus::InputProcessor::factory();
+            }
+
+            return m_inputProcessor.get();
+        }
+    };
+
+    LazyConstructedInputProcessor inputProcessor;
```

Comment:
There's no concern about thread safety here, is there?

Also, just so I understand, this code change stands totally alone, right?  You were able to take advantage of the fact that we already use the spear operator and we can overload it without having to change the code all over the code.  This is nice.

And this is only related to improving the runtime for cases when we don't _actually_ need the input processor.  I could see an argument where we say we should have better design of our unit tests for cases that don't need the input processor to use a better fixture.  But if this change doesn't cause any problems, then it may be fine to drop in.  I'm not strongly opinionated on this one.

Reply:
Let's see - it's not static, so there's no way for one thread to affect another thread *if they are using two different State objects*.

I don't think this changes the situation any from the standpoint of multiple threads access the same E+ objects - which currently does not work, as far as I know.

It does add a tiny extra cost to each access to the inputProcessor. So yes, it could make sense to remove this change since it does only help tests that don't need the inputProcessor.

Reply:
> Also, just so I understand, this code change stands totally alone, right? You were able to take advantage of the fact that we already use the spear operator and we can overload it without having to change the code all over the code. This is nice.

Yeah, should be as simple as just deleting the block of code to go back to how it was.

Reply:
OK, I think I'll revert this change back then and we'll focus on improving the parallel runs by changing the test fixture instead.  Thanks!